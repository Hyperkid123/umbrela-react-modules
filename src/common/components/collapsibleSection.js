import React, {Component} from 'react';
import PropTypes from 'prop-types';
import ExpansionPanel, {
  ExpansionPanelSummary,
  ExpansionPanelDetails,
} from 'material-ui/ExpansionPanel';
import Typography from 'material-ui/Typography';
import ExpandMoreIcon from 'material-ui-icons/ExpandMore';

export default class CollapsibleSection extends Component {
  render() {
    return (
      <ExpansionPanel>
        <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
          <Typography>Expansion Panel 1</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <Typography>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex,
            sit amet blandit leo lobortis eget.
          </Typography>
        </ExpansionPanelDetails>
      </ExpansionPanel>
    );
  }
}

CollapsibleSection.propTypes = {
  sectionTitle: PropTypes.string.isRequired,
  children: PropTypes.object,
};
