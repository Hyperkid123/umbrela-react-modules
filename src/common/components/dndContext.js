import {DragDropContext} from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import React,{Component} from 'react';

export class DNDContext extends Component {
    render() {
        return this.props.children
    }
}



export default DragDropContext(HTML5Backend)(DNDContext);
