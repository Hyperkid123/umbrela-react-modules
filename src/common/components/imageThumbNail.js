import React from 'react';
import {ThumbNail} from '../styledComponents/containers';

export const ImageThumbNail = ({src, alt}) => (
  <div style={{margin: 5, display: 'inline-block'}}>
    <ThumbNail src={src} alt={alt}/>
  </div>
)
