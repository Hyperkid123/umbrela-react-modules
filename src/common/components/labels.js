import React from 'react';
import {
  LabelContainer,
} from '../styledComponents/containers';

export const TextFieldComent = ({label, alignRight, error}) => (
  <LabelContainer alignRight error={error}>
    {label}
  </LabelContainer>
);
