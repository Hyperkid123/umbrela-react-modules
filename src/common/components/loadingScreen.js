import React from 'react';
import { CircularProgress } from 'material-ui/Progress';
import {Flex} from '../styledComponents/containers';

const LoadingScreen = () => (
    <Flex horizontalCenter grow verticalCenter>
      <CircularProgress size={80} thickness={5} />
    </Flex>
);

export default LoadingScreen;
