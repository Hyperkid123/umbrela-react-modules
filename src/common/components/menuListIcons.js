import React from 'react';
import ExpandLess from 'material-ui-icons/ExpandLess';
import ExpandMore from 'material-ui-icons/ExpandMore';
import IconButton from 'material-ui/IconButton';
import {
  MenuListIconsContainer,
  MenuItem,
  MenuListLabel,
  Flex,
} from '../styledComponents/containers';
import {DefaultLink} from '../styledComponents/typography';
import Tooltip from 'material-ui/Tooltip';

export const ChevronUp = ({tooltip}) => (
  <IconButton tooltip={tooltip}>
    <ExpandLess />
  </IconButton>
);

export const ChevronDown = ({tooltip}) => (
  <IconButton tooltip={tooltip}>
    <ExpandMore />
  </IconButton>
);

export const MenuListIconsWrapper = ({tooltipUp, tooltipDown}) => (
  <MenuListIconsContainer>
    <ChevronUp tooltip={tooltipUp}/>
    <ChevronDown tooltip={tooltipDown}/>
  </MenuListIconsContainer>
)

export const MenuListItem = ({link, label, onClick, sheetId, active, tooltipLabel, dragging, preventDrag}) => (
        <Tooltip
          title={tooltipLabel}
          enterDelay={500}
          disableTriggerFocus={dragging || preventDrag}
          disableTriggerHover={dragging || preventDrag}
          disableTriggerTouch={dragging || preventDrag}
          placement='right-end'
        >
          <MenuItem dragging={dragging} data-tip={tooltipLabel} active={active} onClick={() => onClick(sheetId)}>
          <Flex verticalCenter>
          <MenuListLabel>
            <DefaultLink href={link}>{label}</DefaultLink>
          </MenuListLabel>
        </Flex>
    </MenuItem>
  </Tooltip>
)
