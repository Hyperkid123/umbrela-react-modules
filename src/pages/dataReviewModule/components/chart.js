import React,{Component} from 'react';
//import CloseQuestions from './CloseQuestions';
import OpenQuestion from './openQuestion';
//import MatrixQuestion from './MatrixQuestion';
import PropTypes from 'prop-types';
import {Flex} from '../../../common/styledComponents/containers';
//import LegendTable from './LegendTable';
//import DivideQuestion from './DivideQuestion';

import {
  OPEN_QUESTION,
  CLOSE_QUESTION,
  CLOSE_MULTI_QUESTION,
  CLOSE_WITH_OPEN_QUESTION,
  CLOSE_MULTI_WITH_OPEN_QUESTION,
  MATRIX_MULTI_IMAGE_QUESTION,
  MATRIX_MULTI_QUESTION,
  MATRIX_SINGLE_IMAGE_QUESTION,
  MATRIX_SINGLE_QUESTION,
  DIVIDE_QUESTION,
  ORDER_QUESTION,
  OPEN_WITH_IMAGE_QUESTIONS,
  IMAGE_OPTIONS_QUESTION,
  IMAGE_OPTIONS_MULTI_QUESTION,
  IMAGE_OPTIONS_WITH_OPEN_QUESTION,
  IMAGE_OPTIONS_MULTI_WITH_OPEN_QUESTION,
  MEDIA_QUESTION,
  MEDIA_MULTI_QUESTION,
  MEDIA_WITH_OPEN_QUESTION,
  MEDIA_MULTI_WITH_OPEN_QUESTION,
  isOptionImage,
  hasImagePreview,
} from '../../../common/chartTypes';

export default class Chart extends Component {

  renderChart = () => {
    switch (this.props.questionType) {
      case OPEN_QUESTION:
      case OPEN_WITH_IMAGE_QUESTIONS:
        return (
          <Flex>
            <OpenQuestion question={this.props.question} data={this.props.data}/>
          </Flex>
        );
      case CLOSE_QUESTION:
      case CLOSE_MULTI_QUESTION:
      case CLOSE_WITH_OPEN_QUESTION:
      case CLOSE_MULTI_WITH_OPEN_QUESTION:
      case IMAGE_OPTIONS_QUESTION:
      case IMAGE_OPTIONS_MULTI_QUESTION:
      case IMAGE_OPTIONS_WITH_OPEN_QUESTION:
      case IMAGE_OPTIONS_MULTI_WITH_OPEN_QUESTION:
      case MEDIA_QUESTION:
      case MEDIA_MULTI_QUESTION:
      case MEDIA_WITH_OPEN_QUESTION:
      case MEDIA_MULTI_WITH_OPEN_QUESTION:
        return (
            <Flex>
              {/**<CloseQuestions
                hasImagePreview={hasImagePreview(this.props.question.type)}
                showLegend={this.props.showLegend}
                chartType={this.props.chartType}
                data={this.props.data.barData}
                url={this.props.question.url}
              />
              <LegendTable isOptionImage={isOptionImage(this.props.question.type)} data={this.props.data}/>
            */}
            </Flex>
        );
      case  MATRIX_MULTI_IMAGE_QUESTION:
      case  MATRIX_MULTI_QUESTION:
      case  MATRIX_SINGLE_IMAGE_QUESTION:
      case  MATRIX_SINGLE_QUESTION:
        return (
          <Flex>
            {/**

            <MatrixQuestion
              data={this.props.data}
              chartType={this.props.chartType}
              showLegend={this.props.showLegend}
              question={this.props.question}
              isOptionImage={isOptionImage(this.props.question.type)}
            />
          */}
          </Flex>
        );
      case DIVIDE_QUESTION:
      case ORDER_QUESTION:
        return(
          {/**<DivideQuestion data={this.props.data}/>*/}
        );
      default:
        return null;

    }
  }

  render() {
    return this.renderChart();
  }
}

Chart.propTypes = {
  data: PropTypes.object.isRequired,
  chartType: PropTypes.string.isRequired,
  showLegend: PropTypes.bool.isRequired,
  questionType: PropTypes.string.isRequired,
  question: PropTypes.object.isRequired,
};
