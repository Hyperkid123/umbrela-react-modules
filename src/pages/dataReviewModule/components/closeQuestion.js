import React,{Component} from 'react';
import {Doughnut, Bar, Polar, Line} from 'react-chartjs-2';
import PropTypes from 'prop-types';
import {Flex, ChartContainer} from '../../../common/styledComponents/containers';
import Dialog from 'material-ui/Dialog';
import ImageIcon from 'material-ui-icons/Image';
import Button from 'material-ui/Button';

import {
   PIE_CHART,
   DOUGHNUT_CHART,
} from '../../../common/chartTypes';



export default class CloseQuestions extends Component {

  constructor(props){
    super(props);
    this.state = {
      showImage: false,
    };
  }

  handleShowImage = () => this.setState({showImage: true});
  handleCloseImage = () => this.setState({showImage: false});

  renderChart = () => {
    const globalOptions = {
      responsive: true,
      legend: {
        display: !this.props.showLegend,
      },

      tooltips: {
        callbacks: {
          title: function (tooltipItem, data) {
            return data.labels[tooltipItem[0].index];
          },
          label: function (tooltipItem, data) {
            var amount = data.datasets[tooltipItem.datasetIndex].data[tooltipItem.index];
            var total = eval(data.datasets[tooltipItem.datasetIndex].data.join('+'));
            return amount + ' / ' + total + ' ( ' + parseFloat(amount * 100 / total).toFixed(2) + '% )';
          },
        },
      },
    };

    const charts = {
      PIE_CHART: <Doughnut options={{...globalOptions, cutoutPercentage: 0}} width={400} height={400} data={this.props.data.data} redraw={true}/>,
      BAR_CHART: <Bar options={globalOptions} data={this.props.data.data} redraw={true}/>,
      POLAR_CHART:<Polar options={globalOptions} data={this.props.data.data} redraw={true}/>,
      LINE_CHART:<Line options={globalOptions} data={this.props.data.data} redraw={true}/>,
      DOUGHNUT_CHART:<Doughnut options={globalOptions} data={this.props.data.data} redraw={true}/>,
    };
    return charts[this.props.chartType];
  }

  renderImagePreview = () => {
    if(this.props.url){
      return(
         <Button
           fab
           onClick={this.handleShowImage}
           color='primary'
           style={{position: 'absolute'}}
         >
           <ImageIcon />
         </Button>
      );
    }
    return null;
  }

  render() {
    return (
      <Flex style={{position: 'relative'}}>
        <ChartContainer horizontalCenter pie={this.props.chartType === PIE_CHART || this.props.chartType === DOUGHNUT_CHART}>
          {this.renderChart()}
        </ChartContainer>
        {this.renderImagePreview()}
        <Dialog
          open={this.state.showImage}
          onClose={this.handleCloseImage}
        >
          <Flex>
            <img src={this.props.url}/>
          </Flex>
        </Dialog>
      </Flex>

    );
  }
}

CloseQuestions.propTypes = {
  data: PropTypes.object.isRequired,
  chartType: PropTypes.string.isRequired,
  showLegend: PropTypes.bool.isRequired,
  noAnimation: PropTypes.bool,
  url: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
};
