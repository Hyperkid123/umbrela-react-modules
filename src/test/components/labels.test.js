import React from 'react'
import {TextFieldComent} from '../../common/components/labels';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('collapsibleSection component' , () => {
  it('should match snapshot', () => {
    const tree = shallow(
      <TextFieldComent label='foo'/>
    );
    expect(toJson(tree)).toMatchSnapshot();
  })
})
