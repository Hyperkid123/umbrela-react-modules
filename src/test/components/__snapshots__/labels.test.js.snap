// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapsibleSection component should match snapshot 1`] = `
<styled.div
  alignRight={true}
>
  foo
</styled.div>
`;
